pkgs/development/compilers/ghc/9.2.4.nix

```
@@ -194,6 +194,16 @@ stdenv.mkDerivation (rec {
       extraPrefix = "utils/haddock/";
       stripLen = 1;
     })
+    # Don't generate code that doesn't compile when --relocatable is passed to
+    # Setup.hs
+    # Can be removed if the Cabal library included with ghc backports the linked fix
+    (fetchpatch {
+      url = "https://github.com/haskell/cabal/commit/6c796218c92f93c95e94d5ec2d077f6956f68e98.patch";
+      sha256 = "sha256-yRQ6YmMiwBwiYseC5BsrEtDgFbWvst+maGgDtdD0vAY=";
+
+      stripLen = 1;
+      extraPrefix = "libraries/Cabal/";
```

```suggestion
      stripLen = 1;
      extraPrefix = "libraries/Cabal/";
      sha256 = "sha256-yRQ6YmMiwBwiYseC5BsrEtDgFbWvst+maGgDtdD0vAY=";
```
FYI the FOD hash is dependend on those other options

---

Done, Thanks.